src/Functions/HasTokenImpl.h

```diff
@@ -46,10 +47,24 @@ struct HasTokenImpl
         /// The current index in the array of strings.
         size_t i = 0;
 
-        TokenSearcher searcher(pattern.data(), pattern.size(), end - pos);
+        std::optional<TokenSearcher> searcher;
+        try
+        {
+            searcher.emplace(pattern.data(), pattern.size(), end - pos);
```

Comment:
Could you please check if with "OrNull" semantics the actual search (l. 67) should output NULL and continue? Asking because this is the semantics for conversion "OrNull" functions --> https://clickhouse.com/docs/en/sql-reference/functions/type-conversion-functions/ 

Please add comments about this to the code, it is not immediately clear to the reader.

Reply:
I think I know what you mean and I do agree that it can be confusing. In this particular case, the `needle` parameter is supposed to be a literal and an exception from the constructor of `TokenSearcher` would be thrown otherwise. If no exception is thrown at that point, then no further error cases may occur as far as I could tell.

Reply:
I am fine with that, but please add a comment to the code. Thanks!

Reply:
I'll do that, thanks!